### PR TITLE
Migrate deck-drawer to TS

### PR DESCRIPTION
**Pull request by Thaoden**
Migrate `deck-drawer.js` to TypeScript.

I had to move it to a `.tsx` file instead of a `.ts` file in order to use the `<></>` syntax (`DeckDrawer.tsx`, line 43), but I'm not quite sure if I'm using it as intended...

### CODE REVIEW

**Comment by Thaoden** [comment]
Will do that!

File: src/window_main/deck-details.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { MANA, CARD_RARITIES, EASING_DEFAULT } from "../shared/constants";
 import db from "../shared/database";
 import pd from "../shared/player-data";
 import { createDiv, createSpan, queryElements as $$ } from "../shared/dom-fns";
-import * as deckDrawer from "../shared/deck-drawer";
+import * as deckDrawer from "../shared/DeckDrawer";
```
Comment (lusbenjamin) [inline src/window_main/deck-details.js]:
```suggestion
import * as deckDrawer from "./DeckDrawer";
```

File: src/window_main/match-details.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { MANA, EASING_DEFAULT } from "../shared/constants";
 import db from "../shared/database";
 import pd from "../shared/player-data";
 import { createDiv, queryElements as $$ } from "../shared/dom-fns";
-import * as deckDrawer from "../shared/deck-drawer";
+import * as deckDrawer from "../shared/DeckDrawer";
```
Comment (lusbenjamin) [inline src/window_main/match-details.js]:
```suggestion
import * as deckDrawer from "./DeckDrawer";
```

File: src/window_main/renderer-util.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ import {
   createSpan,
   queryElements as $$
 } from "../shared/dom-fns";
-import * as deckDrawer from "../shared/deck-drawer";
+import * as deckDrawer from "../shared/DeckDrawer";
```
Comment (lusbenjamin) [inline src/window_main/renderer-util.js]:
```suggestion
import * as deckDrawer from "./DeckDrawer";
```

File: src/window_main/settings.js [hunk 4]
```diff
@@ -33,7 +33,7 @@ import {
   createLabel,
   queryElements as $$
 } from "../shared/dom-fns";
-import * as deckDrawer from "../shared/deck-drawer";
+import * as deckDrawer from "../shared/DeckDrawer";
```
Comment (lusbenjamin) [inline src/window_main/settings.js]:
```suggestion
import * as deckDrawer from "./DeckDrawer";
```

File: src/window_main/tournaments.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import anime from "animejs";
 import { queryElements as $$, createDiv } from "../shared/dom-fns";
 import createSelect from "./createSelect";
-import * as deckDrawer from "../shared/deck-drawer";
+import * as deckDrawer from "../shared/DeckDrawer";
```
Comment (lusbenjamin) [inline src/window_main/tournaments.js]:
```suggestion
import * as deckDrawer from "./DeckDrawer";
```

**Comment by Thaoden** [comment]
Uh, looks like I missed some dependencies...

**Pull request merged**